src/oas3/__tests__/bundle.test.ts

```diff
@@ -1289,4 +1290,35 @@ describe('bundleOas3Service', () => {
       ]);
     });
   });
+
+  it('when an operation-level param ref points outside the http-spec document', () => {
+    const res = bundleOas3Service({
+      document: {
+        openapi: '3.0.0',
+        paths: {
+          '/': {
+            get: {
+              operationId: 'the_op_id',
+              responses: { '200': {} },
+              parameters: [
+                {
+                  // BUG: This parameter is missing from the http-spec document
+                  $ref: 'otherFile.yaml#/components/parameters/Shared1',
+                },
+              ],
+            },
+          },
+        },
+      },
+    });
+
+    console.log(JSON.stringify(res, null, 4));
+    const request = res.operations[0].request;
+    if (!request || isReferenceObject(request)) fail('should be a response');
+    expect(request.unknown).toStrictEqual([
+      {
+        $ref: 'otherFile.yaml#/components/parameters/Shared1',
+      },
+    ]);
+  });
```

Comment:
@daniel-white , of the failing tests you pointed out, I _think_ this test shows the common problem between them.  I'm afraid I need to sign off for the day (and week), but hopefully this will at least give you a good starting point tomorrow.

Reply:
all is passing now!